styles/style_ptr.h

```diff
@@ -133,14 +140,20 @@ class ChargingStyle : public Style<T> {
 // Get a pointer to class.
 template<class STYLE>
 StyleAllocator StylePtr() {
+  using _TopBaseT  = typename style_base_check_detail::TopBase<STYLE>::type;
+  using _TopBaseCol = decltype(std::declval<_TopBaseT&>().getColor(0));
+  static_assert(color_details::IsOpaqueColor<_TopBaseCol>::value,
+                "\n\n"
+                "*** StylePtr<> error: BASE LAYER MUST BE A SOLID COLOR, NOT TRANSPARENT.\n"
```

Comment:
This might be confusing if the argument to styleptr is not a layer.
How about we move the actual assert into style_base_check_detail, and we'll make one error message if the argument is Layers<> with a transparent base, and another error message if it's something else that is transparent?

That should also give us a little bit more code reuse since this is currently repeated in StyleChargingPtr<>

Reply:
> another error message if it's something else that is transparent

hmm, meaning something like
`AudioFlicker<AlphaL<Blue,Int<16000>>,AlphaL<Red,Int<16000>>>` ?

How about we just leave the single message but remove "layer" from it so it just says "base must" - that covers a transparent style or a Layer.
"StylePtr<> error: BASE MUST BE A SOLID COLOR, NOT TRANSPARENT